### PR TITLE
Linux kernels 2023-12-20

### PR DESCRIPTION
**Pull request by alyssais**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by K900** [comment]
No 6.1/6.6?

**Comment by alyssais** [comment]
Not yet at least, but these are very fresh.

**Comment by K900** [comment]
And there they are.

File: pkgs/os-specific/linux/kernel/hardened/patches.json [hunk 1]
```diff
@@ -1,4 +1,14 @@
 {
+    "4.14": {
```
Comment (K900) [inline pkgs/os-specific/linux/kernel/hardened/patches.json:2]: 4.14 got revived here, not sure if we care?

Comment (alyssais) [reply pkgs/os-specific/linux/kernel/hardened/patches.json]: Dropped it again.

**Review by K900 — APPROVED** [review]
LGTM

**Comment by fabianhjr** [comment]
Hi, anthraxx hardened kernel has just released. Pushing those updates here too and building 6.6.8

**Comment by fabianhjr** [comment]
6.6.8-hardened failed during config with

```
GOT: make: Leaving directory '/build/linux-6.6.8'
error: unused option: DEBUG_CREDENTIALS
warning: unused option: DEBUG_INFO_BTF
warning: unused option: GCC_PLUGIN_STRUCTLEAK
warning: unused option: GCC_PLUGIN_STRUCTLEAK_BYREF_ALL
warning: unused option: INET_DIAG_DESTROY
warning: unused option: INET_MPTCP_DIAG
warning: unused option: INET_RAW_DIAG
warning: unused option: INET_TCP_DIAG
warning: unused option: INET_UDP_DIAG
warning: unused option: IO_STRICT_DEVMEM
warning: unused option: MODULE_ALLOW_BTF_MISMATCH
warning: unused option: SECURITY_WRITABLE_HOOKS
```

**Comment by fabianhjr** [comment]
6.6.8 non hardened doesn't have this issues, apologies for the noise, removing the hardened updates

**Comment by fabianhjr** [comment]
Reset back to same previous head. Builds

**Review by fabianhjr — APPROVED** [review]
LGTM, 6.6.8 builds

**Comment by fabianhjr** [comment]
In a weird github feature/glitch note, apparently approvals are per commit hash and when changing the head back and forth the approval remains.

image

**Comment by K900** [comment]
Can someone please get the 23.05 backport, I'm on my phone.

**Comment by fabianhjr** [comment]
On it

**Comment by fabianhjr** [comment]
missing previous backport:

```
++<<<<<<< HEAD
 +        "version": "5.15.142",
 +        "hash": "sha256:0xjn16b02f8d6c0m8vrbmk85kdyfy8m46s80rnkb0nnwfx9cjxld"
++||||||| parent of e0dabff327fb (linux_5_15: 5.15.143 -> 5.15.144)
++        "version": "5.15.143",
++        "hash": "sha256:00lyv7zsj97mkg9i7dkb1a6km22mnr0qr687d9zz4ckjq1pb2sq9"
++=======
+         "version": "5.15.144",
+         "hash": "sha256:0fsv18q64q17ad7mq818wfhb11dax4bdvbvqyk5ilxyfmypsylzh"
++>>>>>>> e0dabff327fb (linux_5_15: 5.15.143 -> 5.15.144)
      },
      "5.10": {
```

**Comment by fabianhjr** [comment]
https://github.com/NixOS/nixpkgs/pull/275761

**Pull request merged**